multidataset_plugin.cc

```diff
@@ -0,0 +1,544 @@
+#include "multidataset_plugin.h"
+
+
+static multidataset_array *multi_datasets;
+/*
+static hid_t *dataset_recycle;
+static hid_t *memspace_recycle;
+static hid_t *dataspace_recycle;
+*/
+static int dataset_size;
+static int dataset_size_limit;
+
+/*
+static int dataset_recycle_size;
+static int dataset_recycle_size_limit;
+static int dataspace_recycle_size;
+static int dataspace_recycle_size_limit;
+static int memspace_recycle_size;
+static int memspace_recycle_size_limit;
+*/
+hsize_t total_data_size;
+
+#define MEM_SIZE 2048
+#define MAX_DATASET 65536
+
+int init_multidataset() {
+    dataset_size = 0;
+    dataset_size_limit = 0;
+
+    char *p = getenv("HEP_MAX_BATCH_SIZE");
```

Comment:
Please do not use environment variables to tune. These should come from the command line arguments passed in to the program. An easy modification is to pass the values via parameters of this function.